### PR TITLE
SCALRCORE-23827 Add id and name filters for all data sources

### PR DESCRIPTION
**Pull request by denkl**
### Changelog
* [x] I have updated CHANGELOG.md according to [CONTRIBUTING.md](../CONTRIBUTING.md)

### Documentation
* [x] I have updated resource/data source documentation in [docs](../docs)

### State
* [ ] My changes affect the state
* [ ] I have included a state migration and a unit test for it

### CODE REVIEW

File: scalr/data_source_scalr_agent_pool.go [hunk 1]
```diff
@@ -45,17 +46,26 @@ func dataSourceScalrAgentPool() *schema.Resource {
...
 	scalrClient := meta.(*scalr.Client)
-	var envID string
 
+	agentPoolID := d.Get("id").(string)
 	name := d.Get("name").(string)
 	accountID := d.Get("account_id").(string)
+	envID := d.Get("environment_id").(string)
+
 	options := scalr.AgentPoolListOptions{
-		Name:    name,
```
Comment (petroprotsakh) [inline scalr/data_source_scalr_agent_pool.go:53]: Now that the `name` is not required the following configuration gives us the first agent pool in the account, which is kind of weird:
```
data "scalr_agent_pool" "pool" {
  id   = ""
  name = ""
  # one of the id, name can be omitted with the same result
}

output o {
  value = data.scalr_agent_pool.pool
}
```
⬇️
```
GET /agent-pools?filter[account]=acc-***
```
⬇️
```
Changes to Outputs:
  + o = {
      + account_id     = "acc-***"
      + environment_id = null
      + id             = "apool-***"
      + name           = "my-pool"
      + workspace_ids  = null
    }
```

`AtLeastOneOf` loses sense in this case, as the configuration above would be equivalent (in terms of result) to just
```
data "scalr_agent_pool" "pool" {}
```

I'd suggest adding a `StringIsNotEmpty` or `StringIsNotWhiteSpace` validation function for such arguments, hence restricting empty values declaratively at early stage.

Need to check it across all affected datasources.

File: scalr/data_source_scalr_agent_pool.go [hunk 2]
```diff
@@ -15,12 +15,13 @@ func dataSourceScalrAgentPool() *schema.Resource {
 		ReadContext: dataSourceScalrAgentPoolRead,
 		Schema: map[string]*schema.Schema{
 			"id": {
-				Type:     schema.TypeString,
-				Computed: true,
```
[the remark this answers is not in the document]
Comment (denkl) [reply scalr/data_source_scalr_agent_pool.go:19]: Makes sense, added back.

File: scalr/data_source_scalr_agent_pool.go (hunk 1 above)
Comment (denkl) [reply scalr/data_source_scalr_agent_pool.go:53]: Added `StringIsNotWhiteSpace` validation function.

**Review by petroprotsakh — APPROVED** [review]
Looks good

**Pull request merged**